### PR TITLE
Accelerated tracker block #1011

### PR DESCRIPTION
**Pull request by zebesta**
Added a new option to 'Block trackers at first occurrence" and storage to keep track of this option as a boolean. If the user checks this box it will start blocking tracking adds after 1 occurrence instead of 3. Addresses issue #1011 - needs language translation

Could use review for how to word this option for the users, and also, how to submit this for all of the _locales messages.json

image

### CODE REVIEW

**Comment by alexristich** [comment]
@zebesta Awesome first pass at this. Thanks for putting something together so quickly! You're right that the wording needs a bit of work - it's a bit of a tricky thing to explain succinctly :)

@cooperq Here's what I understand the desired functionality to be:
A user is more concerned about their privacy, and would prefer to block all trackers by default rather than waiting for the standard `TRACKING_THRESHOLD` to be reached. With this setting active, when _any_ tracking is detected by a third party origin, the default behavior will be to set the `heurisiticAction` to `cookie_block`.

This part of it sounds great, though I believe we still need to traverse the default path when `TRACKING_THRESHOLD` is reached, else whenever this setting is active no future domains will receive a `heuristicAction` value of `block`. Does that sound about right?

**Comment by ghostwords** [comment]
I suggest against creating disabled-by-default changes to the heuristic. If it's a good change, we should make it for all users, not just a small subset. Off-by-default options could potentially be used as a testing/incremental release strategy, but we should consider the followup process so that it's actually a testing/gradual rollout, and not a rarely used feature we forget about.

**Comment by ghostwords** [comment]
Verifying this change would benefit from having some basic implementation of #1019.

**Comment by zebesta** [comment]
Just curious if looking back in to this issue is worth while based on @ghostwords comment and what the design direction here should be -  Re: Is this a feature that is worth implementing at all if its disabled by default?

If moving forward with this is worthwhile, I could use a little clarification on what your desired functionality is and how it deviates from what is currently implemented with this PR @alexristich @cooperq

**Comment by cooperq** [comment]
@zebesta I think we have decided to take a different direction for now. I am keeping this issue open but  I think it will be dependent on #1019 
After that I still do want to implement it so we should pick it back up then. 
Thanks for the good code though! Please do keep contributing!

**Comment by alexristich** [comment]
@zebesta Also, if you want to get started on another issue feel free to take a look at the Privacy Badger 2.1 milestone (https://github.com/EFForg/privacybadger/milestone/11). Most things there are up for grabs (some being a lot harder than others), so feel free to poke around and see what tickles your fancy!

**Comment by jawz101** [comment]
I say make any heuristic options variables which can be set with a slider or something.
Like, I could slide it to 1,2,3, etc.
Makes me think of Internet Explorer Tracking Protection's Personalized List feature

www.herongyang.com/Windows/IE_10_Tracking_Protection_Personalized_List.jpg

If I ever have to use IE I've always enabled its Tracking Protection Personalized List, set to automatically block, and dialed it down to 3.

Basically, they've had a simple Privacy Badger since I.E. version 9 (since 2011.)  People just didn't really notice it.

**Comment by paoletto** [comment]
what does Block Trackers at First Occurrence mean? Red or Yellow?

**Comment by ghostwords** [comment]
Closing as per https://github.com/EFForg/privacybadger/issues/1011#issuecomment-337636975.

**Comment by paoletto** [comment]
I think this option would have been convenient to test the new behavior, before removing it after assessing whether it makes sense to have the new behavior or not
